### DISCUSSION TITLE
`Internal._fixNGon()` convertir un arbre PyTree NGON4 en NGON3 ?

### DISCUSSION THREAD

**Discussion by Enn83** [onera/Cassiopee#249] [Q&A] [answered]
Bonjour,

J'utilise actuellement Cassiopée 4.0a et lors de l'exécution de la fonction suivante sur un arbre PyTree `t` en **NGON4** :
```py
4624: def _fixNGon(t, remove=False, breakBE=True, convertMIXED=True, addNFace=True):
```
... l'arbre qui en résulte est à nouveau en format **NGON3**. Une conversion est en effet réalisée juste à la fin de la fonction :
```py
4809:   _adaptNGon42NGon3(t)
```

Est-ce normal/nécessaire ?

Cordialement,
Ξии

**Comment by vincentcasseau** [comment, marked as answer]
Bonjour,

Merci pour votre retour d'expérience.
La fonction `_fixNGon` est quelque peu obsolète, il faudrait qu'elle soit retravaillée suite à l'introduction de la norme NGON4 dans Cassiopée. C'est donc le comportement anticipé mais ce n'est ni nécessaire ni normal dorénavant.

C'est un point bloquant par ailleurs, donc on va voir ce qu'on peut faire pour corriger ça dans les semaines à venir.

Cordialement,
Vincent

**Reply by Enn83** [reply]
D'accord, je vais commenter cet appel le temps de faire mes tests et en attendant que la version finale de Cassiopée 4.0 sorte.

Merci beaucoup pour votre réponse et bonne année 2025 🎉 !
